crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -759,6 +767,70 @@ impl Starknet {
         state_update::state_update_by_block_id(self, block_id)
     }
 
+    pub fn abort_blocks(&mut self, starting_block_hash: Felt) -> DevnetResult<Vec<Felt>> {
+        if self.config.state_archive != StateArchiveCapacity::Full {
+            return Err(Error::UnsupportedAction {
+                msg: ("The abort blocks feature requires state-archive-capacity set to full."
+                    .to_string()),
+            });
+        }
+
+        let mut next_block_to_abort_hash = match self.blocks.last_block_hash {
+            None => {
+                return Err(Error::UnsupportedAction {
+                    msg: ("last_block_hash is not set".to_string()),
+                });
+            }
+            Some(felt) => felt,
+        };
```

Comment:
Can be simplified as this (also I think the error message could be change because users don't have to know anything about the `last_block_hash` variable):
```suggestion
        let mut next_block_to_abort_hash = self
            .blocks
            .last_block_hash
            .ok_or(Error::UnsupportedAction { msg: "No blocks to abort".into() })?;
```

Reply:
Fixed in ace808fe050d55afaece868d2854ba6c0ba3d68a